MkIfex.thy

```diff
@@ -7,70 +7,30 @@ theory MkIfex
   imports "ROBDD.BDT"
 begin
 
-fun mk_ifex :: "'a boolfunc \<Rightarrow> 'a list \<Rightarrow> 'a ifex" where
-"mk_ifex f [] = (if f undefined then Trueif else Falseif)" |
-"mk_ifex f (v#vs) = IF v (mk_ifex (bf_restrict v True f) vs) (mk_ifex (bf_restrict v False f) vs)"
+fun mk_ifex :: "('a :: linorder) boolfunc \<Rightarrow> 'a list \<Rightarrow> 'a ifex" where
+"mk_ifex f [] = (if f (const False) then Trueif else Falseif)" |

```

Comment:
Replacing `undefined` by `(const false)` here makes sure that `value` and `eval` will work.